gno.land/pkg/gnoclient/client_txs.go

```diff
@@ -106,59 +102,73 @@ func (c *Client) Call(cfg BaseTxCfg, msgs ...MsgCall) (*ctypes.ResultBroadcastTx
 	return c.signAndBroadcastTxCommit(tx, cfg.AccountNumber, cfg.SequenceNumber)
 }
 
-// Temporarily load cfg.Package on the blockchain and run main() which can
-// call realm functions and use println() to output to the "console".
-// This returns bres where string(bres.DeliverTx.Data) is the "console" output.
-func (c *Client) Run(cfg RunCfg) (*ctypes.ResultBroadcastTxCommit, error) {
-	// Validate required client fields.
+func (c *Client) Run(cfg BaseTxCfg, msgs ...MsgRun) (*ctypes.ResultBroadcastTxCommit, error) {
+	// Validate required client fields
 	if err := c.validateSigner(); err != nil {
-		return nil, errors.Wrap(err, "validate signer")
+		return nil, err
 	}
 	if err := c.validateRPCClient(); err != nil {
-		return nil, errors.Wrap(err, "validate RPC client")
+		return nil, err
 	}
 
-	memPkg := cfg.Package
-	gasWanted := cfg.GasWanted
-	gasFee := cfg.GasFee
-	sequenceNumber := cfg.SequenceNumber
-	accountNumber := cfg.AccountNumber
-	memo := cfg.Memo
-
-	// Validate config.
-	if memPkg.IsEmpty() {
-		return nil, errors.New("found an empty package " + memPkg.Path)
+	// Validate base transaction config
+	if err := cfg.validateBaseTxConfig(); err != nil {
+		return nil, err
 	}
 
-	// Parse gas wanted & fee.
-	gasFeeCoins, err := std.ParseCoin(gasFee)
-	if err != nil {
-		return nil, errors.Wrap(err, "parsing gas fee coin")
+	// Parse MsgRun slice
+	vmMsgs := make([]vm.MsgRun, 0, len(msgs))
```

Comment:
Can we not have this straight as a `[]std.Msg` instead of re-allocating another slice later?

Reply:
True. Fixed it in both MsgRun & MsgCall: [b27064e](https://github.com/gnolang/gno/pull/1637/commits/b27064e8ed64fe2ced5dd683ded2f92573a47979)